graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/DataTypeGenerator.kt

```diff
@@ -360,6 +381,35 @@ abstract class BaseDataTypeGenerator(
 
     private fun addField(fieldDefinition: Field, javaType: TypeSpec.Builder) {
         addFieldWithGetterAndSetter(fieldDefinition.type, fieldDefinition, javaType)
+        // Generate for all fields except non-nullable without default values
+        if (!(!fieldDefinition.isNullable && fieldDefinition.initialValue == null)) {
+            addIsDefinedFieldWithGetters(fieldDefinition, javaType)
+        }
+    }
+
+    private fun addIsDefinedFieldWithGetters(fieldDefinition: Field, javaType: TypeSpec.Builder) {
+        val fieldName = generateBooleanFieldName(ReservedKeywordSanitizer.sanitize(fieldDefinition.name))
+        val field = FieldSpec
+            .builder(com.squareup.javapoet.TypeName.BOOLEAN, fieldName)
+            .addModifiers(Modifier.PRIVATE)
+            .initializer("false")
+            .build()
+        val getterName = "${fieldName}Defined"
+
+        val getter = MethodSpec
+            .methodBuilder(getterName)
+            .addModifiers(Modifier.PUBLIC)
+            .returns(com.squareup.javapoet.TypeName.BOOLEAN)
+            .addStatement(
+                "return \$N",
+                "${generateBooleanFieldName(ReservedKeywordSanitizer.sanitize(fieldDefinition.name))}"
```

Comment:
String interpolation is not needed here, just pass the result of `generateBooleanFieldName` to the method.

Reply:
Since field names here are of internal type `Field`, removing string interpolation generates incorrect field names for test cases where java keywords are used as field names: 
1. Should be able to generate a valid client when java keywords are used as field names
2. Should be able to generate successfully when java keywords and default value are used as input types
3. Should be able to generate successfully when java keywords are used as types

Reply:
I'm talking about changing that line from this:

```kotlin
"${generateBooleanFieldName(ReservedKeywordSanitizer.sanitize(fieldDefinition.name))}"
```

to this:

```kotlin
generateBooleanFieldName(ReservedKeywordSanitizer.sanitize(fieldDefinition.name))
```

IntelliJ even shows a warning for this, since the String interpolation isn't necessary. I have no idea how that change would break anything, it doesn't for me locally.

And actually now that I look at it, there's already a `fieldName` variable that you can just use here instead.

Reply:
Thanks, updated 👍